File: Solutions/PAT/Basic/1001.rb
#!/usr/bin/env ruby
# 害死人不偿命的(3n+1)猜想 (15)

x = gets.to_i
cnt = 0
while x != 1 do
  x = (x & 1 == 1 and (3 * x + 1) >> 1 or x >> 1)
  cnt += 1
end
puts cnt

File: Solutions/PAT/Basic/1001.lua
#!/usr/bin/env lua
-- 害死人不偿命的(3n+1)猜想 (15)

n = io.read("*number")
cnt = 0
while n ~= 1 do
  n = (n & 1 == 0) and (n >> 1) or ((3 * n + 1) >> 1)
  cnt = cnt + 1
end
print(math.floor(cnt))